src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -1154,27 +1151,215 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
                 }
 
 #if defined(TARGET_XARCH)
-                case NI_VectorT_ConvertToInt32:
+
+                case NI_VectorT_ConvertToInt64:
```

Comment:
Ideally we'd have some `gtNewSimdConvertNode` helper that both `simdashwintrinsic` and `hwintrinsicxarch` could defer to, that way the code is shared and not duplicated in two places.

Reply:
Sure. Let me make those changes. I was thinking of a good way to handle the control table but we can do that with this PR as well. I will make appropriate changes where needed.

Reply:
This review was incorporated in out latest changes.